### PR TITLE
fix: bridging tx's statuses

### PR DESCRIPTION
**Pull request by skylarbarrera**
Fixes APP-80

## What changed (plus any additional context for devs)

handling regarding bridge tx statuses was kinda wonky 👇 

1. we we're checking token approvals via the socket api, this isn't handled by them and was causing issues. now we just check like we do a normal tx
2. if we got any type of error from socket we were marking the tx as failed -> their api seems kinda wonky atm and would throw errors when it shouldn't. we should only mark as failed if socket explicitly tells us the tx failed, otherwise we should continue watching the tx status
^ ive kept the `IS_TEST` logic as this api wont work on e2e tests since they're simulated tx's

## Screen recordings / screenshots
https://cloud.skylarbarrera.com/Screen-Recording-2022-11-14-14-05-22.mp4

## What to test
imma let ibrahim do his thing on this one so code review is 👌

### CODE REVIEW

File: src/redux/data.ts [hunk 1]
```diff
@@ -1396,10 +1395,18 @@ export const dataWatchPendingTransactions = (
...
+            transactionStatus === TransactionStatus.approved ||
+            transactionStatus === TransactionStatus.approving;
+          if (
+            updatedPendingTransaction?.swap?.type === SwapType.crossChain &&
+            !isApproveTx
+          ) {
             pendingTransactionData = await getTransactionSocketStatus(
               updatedPendingTransaction
             );
+            logger.debug('pending tx data: ', pendingTransactionData);
```
Comment (benisgold) [inline src/redux/data.ts:1409]: switch to new logger?

**Review by benisgold — APPROVED** [review]

File: src/redux/data.ts (hunk 1 above)
Comment (skylarbarrera) [reply src/redux/data.ts:1409]: ah yes I was going to but got scared with all the obj we're logging, will revisit

**Review by ibrahimtaveras00 — CHANGES_REQUESTED** [review]
There's an issue where the text is blank when sending a txn and you just see the spinner on both OS's 

seen here = https://linear.app/rainbow/issue/APP-80#comment-367ad389

File: src/redux/data.ts (hunk 1 above)
Comment (estrattonbailey) [reply src/redux/data.ts:1409]: Yeah this seems like an OK `debug`. Not sure we need this in Sentry, but might be useful locally 👍

Comment (estrattonbailey) [reply src/redux/data.ts:1409]: Object is correct in this case, it'll just be pretty-printed to the console.

**Review by estrattonbailey — COMMENTED** [review]
Just the logs, otherwise looks great 👍

File: src/redux/data.ts [hunk 2]
```diff
@@ -1360,7 +1359,7 @@ export const dataWatchPendingTransactions = (
         status: TransactionStatus;
       } | null = {
         status: TransactionStatus.sending,
-        title: '',
+        title: pendingTransactionData?.title || TransactionStatus.sending,
```
Comment (skylarbarrera) [inline src/redux/data.ts:1362]: this was causing empty tittles if nothing happens in the logic below

we should use the same title we has previously and if its not there default to a generic send

**Review by ibrahimtaveras00 — APPROVED** [review]
Fixes look good on both OS's, QA Passed 👍🏽

**Pull request merged**